src/Text/Pandoc/Writers/XWiki.hs

```diff
@@ -0,0 +1,256 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-
+Copyright (C) 2008-2017 John MacFarlane <jgm@berkeley.edu>
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 2 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program; if not, write to the Free Software
+Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+-}
+
+{- |
+   Module      : Text.Pandoc.Writers.XWiki
+   Copyright   : Copyright (C) 2008-2017 John MacFarlane
+   License     : GNU GPL, version 2 or above
+
+   Maintainer  : Derek Chen-Becker <dchenbecker@gmail.com>
+   Stability   : alpha
+   Portability : portable
+
+Conversion of 'Pandoc' documents to XWiki markup.
+
+XWiki:  <http://www.xwiki.org/>
+XWiki Syntax:  <http://www.xwiki.org/xwiki/bin/view/Documentation/UserGuide/Features/XWikiSyntax/>
+-}
+
+module Text.Pandoc.Writers.XWiki ( writeXWiki ) where
+import Control.Monad.Reader (ReaderT, asks, local, runReaderT)
+import Data.Monoid
+import qualified Data.Set as Set
+import qualified Data.Text as Text
+import Data.Text (Text, concat, init, intercalate, pack, replace, split, unlines, unwords)
+import Text.Pandoc.Class (PandocMonad, report)
+import Text.Pandoc.Definition
+import Text.Pandoc.Logging
+import Text.Pandoc.Options
+import Text.Pandoc.Shared (escapeURI, isURI, linesToPara)
+import Text.Pandoc.Writers.MediaWiki (highlightingLangs)
+
+data WriterState = WriterState {
+  listLevel :: Text -- String at the beginning of items
+}
+
+type XWikiReader m = ReaderT WriterState m
+
+-- | Convert Pandoc to XWiki.
+writeXWiki :: PandocMonad m => WriterOptions -> Pandoc -> m Text
+writeXWiki _ (Pandoc _ blocks) =
+  let env = WriterState { listLevel = "" }
+  in do
+  body <- runReaderT (blockListToXWiki blocks) env
+  return $ body
+
+-- | Concatenates strings with line breaks between them.
+vcat :: [Text] -> Text
+vcat = intercalate "\n"
+
+blockListToXWiki :: PandocMonad m => [Block] -> XWikiReader m Text
+blockListToXWiki blocks =
+  fmap vcat $ mapM blockToXWiki blocks
+
+blockToXWiki :: PandocMonad m => Block -> XWikiReader m Text
+
+blockToXWiki Null = return ""
+
+-- TODO: handle this
```

Comment:
What handling does this need?
If there is a way to mark internal anchors in xwiki, one could check the id attribute and set an anchor.

Reply:
I've added this, but there doesn't seem to be a test case in testsuite.native where a Div has an id set (they're all ""):

```
9 matches for "Div" in buffer: testsuite.native
    231:,Div ("",[],[])
    234:,Div ("",[],[])
    235: [Div ("",[],[])
    236:  [Div ("",[],[])
    238: ,Div ("",[],[])
    253:,Div ("",[],[])
    260:,Div ("",[],[])
    261: [Div ("",[],[])
    262:  [Div ("",[],[])
```

Reply:
OK, confirmed this is working with a quick `text.html` file:

```html
<div id="test">
This is div 3
</div>

<a href="#test">Link to test div</a>
```

produces:

```xwiki
{{ id name="test" /}}This is div 3
[[Link to test div>>#test]]
```